src/rich-text/plugins/link-paste-handler.ts

```diff
@@ -0,0 +1,78 @@
+import { Plugin, EditorState } from "prosemirror-state";
+import { EditorView } from "prosemirror-view";
+import { Node, Schema } from "prosemirror-model";
+import { validateLink } from "../../shared/utils";
+
+function isInlineCode(state: EditorState): boolean {
+    const { from, $from, to, empty } = state.selection;
+    const schema = state.schema as Schema;
+    if (!empty) {
+        return state.doc.rangeHasMark(from, to, schema.marks.code);
+    }
+
+    return (
+        schema.marks.code.isInSet(state.storedMarks || $from.marks()) !==
+        undefined
+    );
```

Comment:
nit: `isInSet` could also return `null`. We can catch both cases by doing the following:

```suggestion
    return !!schema.marks.code.isInSet(state.storedMarks || $from.marks());
```

Reply:
Side note: Looks like we have this (almost exact) code in a few places now - I'll extract [this function](https://github.com/StackExchange/Stacks-Editor/blob/main/src/rich-text/commands.ts#L578) into utils and update this code to match later.

No changes necessary.